crates/cairo-profiler/src/trace_reader/function_trace_builder.rs

```diff
@@ -113,7 +114,8 @@ pub fn collect_profiling_info(
             }
         };
 
-        let user_function_idx = user_function_idx(sierra_statement_idx, program);
+        let user_function_name = user_function_name(sierra_statement_idx, program)
+            .to_displayable_function_name(function_level_config.split_generics);
```

Comment:
Passing the `split_generics` here may seem weird but it actually makes sense. If a users choses not to split them then e.g. `func<felt252>` and `func<u8>` look the same to them, so if they are called consecutively one after the other they should be aggregated for the purpose of function stack depth - they see them as a single function on the stack anyways.

Reply:
I generally think the problem here is that you pass those dependencies around. You should have a struct that holds program and split_generics, some kind of database, and perform a query with `sierra_statement_idx` on that.